File: src/2015/Day_01/P2.rb
data = File.read("input.txt").split("")
floor = 0
result = 0

i = 0
for token in data
    if token === "("
        floor += 1
    else
        floor -= 1
    end

    if result === 0 && floor === -1
        result = i + 1
    end
    i += 1
end

puts "Result: #{result}"

File: src/2022/Day 01/P2.js
const fs = require(`fs`);
const path = require(`path`);

const main = async () => {
    fs.readFile(path.resolve(__dirname, `./input.txt`), `utf-8`, (err, data) => {
        if (err) throw err;

        const lines = data.split(`\n\r`).map(x => x.trim());
        let ans = 0;

        const totals = [];
        for (const line of lines) {
            const cals = line.split(`\n`).map(x => parseInt(x.trim()));
            totals.push(cals.reduce((a, b) => a + b));
        }

        totals.sort((a, b) => b - a);

        console.log(`Result: ${totals[0] + totals[1] + totals[2]}`);
        fs.writeFileSync(path.resolve(__dirname, `./output.txt`), String(ans));
    });
};

void main();
